robotoff/app/core.py

```diff
@@ -126,6 +126,8 @@ def get_insights(
         elif order_by == "n_votes":
             query = query.order_by(ProductInsight.n_votes.desc())
 
+    query = query.order_by(ProductInsight.timestamp.desc())
+
```

Comment:
you should add this only if we have a order_by == "most_recent" parameter, not always, because you are going to change existing code in an unpredictable manner.

This is what makes the test failed by the way (because your line removes previous ordering).